observ/watcher.py

```diff
@@ -123,10 +123,13 @@ def __init__(
         callback: Method to call when value has changed
         """
         self.id = next(_ids)
-        if isinstance(fn, Proxy):
-            self.fn = lambda: fn
-        else:
+        if callable(fn):
             self.fn = fn
+        else:
+            self.fn = lambda: fn
+            # When watching proxies or a list of proxies
+            # make sure that deep watching is enabled
+            deep = True
```

Comment:
The default should be True, but now it's impossible to set it to False for the user

Reply:
Ah, indeed. Good point about setting the default value to None